hack/verify-golint.sh

```diff
@@ -24,7 +24,8 @@ readonly KUBE_ROOT=$(dirname "${BASH_SOURCE}")/..
 cd "${KUBE_ROOT}"
 
 # See configuration file in ${KUBE_ROOT}/.golangci.yml.
+mkdir -p cache
 
-docker run --rm -v $(pwd):/app -w /app "golangci/golangci-lint:$VERSION" golangci-lint run
+docker run --rm -v $(pwd)/cache:/cache -v $(pwd):/app -e GOLANGCI_LINT_CACHE=/cache -w /app "golangci/golangci-lint:$VERSION" golangci-lint run
```

Comment:
This actually doesn't solve the problem.
It downloads all of the modules again, which then leads to re-computation from scratch.

Reply:
Really? I thought that the cache would be reused between runs. It definitely saves time though, or I wouldn't have suggested it.

Reply:
Yes, running with the cache means that you save some time (about 40s on my MacBook) by not reanalysing:
With an empty cache:
```bash
─❯  ./hack/verify-golint.sh
level=info msg="[config_reader] Config search paths: [./ /app / /root]"
level=info msg="[config_reader] Used config file .golangci.yml"
level=info msg="[lintersdb] Active 24 linters: [asciicheck bodyclose deadcode errcheck errorlint exhaustive exportloopref gofmt goimports golint gosec govet ineffassign misspell noctx rowserrcheck scopelint sqlclosecheck staticcheck structcheck typecheck unparam unused varcheck]"
level=info msg="[lintersdb] Active presets: [bugs unused]"
level=info msg="[loader] Go packages loading at mode 575 (compiled_files|deps|files|name|exports_file|imports|types_sizes) took 44.256566822s"
level=info msg="[runner/filename_unadjuster] Pre-built 0 adjustments in 9.704731ms"
level=info msg="[linters context/goanalysis] analyzers took 39.487204519s with top 10 stages: buildir: 30.938890346s, buildssa: 2.239072216s, inspect: 1.254830899s, exhaustive: 1.195800486s, goimports: 914.249541ms, ctrlflow: 618.427116ms, printf: 559.238695ms, fact_purity: 444.588324ms, unparam: 312.92591ms, fact_deprecated: 263.153064ms"
level=info msg="[linters context/goanalysis] analyzers took 788.288438ms with top 10 stages: buildir: 743.926061ms, U1000: 44.362377ms"
level=info msg="[runner] Issues before processing: 122, after processing: 0"
level=info msg="[runner] Processors filtering stat (out/in): exclude: 47/47, filename_unadjuster: 122/122, skip_files: 122/122, autogenerated_exclude: 47/122, path_prettifier: 122/122, exclude-rules: 0/47, cgo: 122/122, skip_dirs: 122/122, identifier_marker: 47/47"
level=info msg="[runner] processing took 4.250246ms with stages: identifier_marker: 1.40125ms, autogenerated_exclude: 1.255265ms, path_prettifier: 1.134776ms, skip_dirs: 219.366µs, exclude-rules: 199.8µs, cgo: 23.436µs, filename_unadjuster: 10.794µs, nolint: 1.234µs, max_same_issues: 771ns, skip_files: 564ns, exclude: 478ns, uniq_by_line: 453ns, source_code: 349ns, max_from_linter: 337ns, path_shortener: 299ns, diff: 279ns, sort_results: 247ns, severity-rules: 246ns, max_per_file_from_linter: 161ns, path_prefixer: 141ns"
level=info msg="[runner] linters took 24.562060908s with stages: goanalysis_metalinter: 23.538912398s, unused: 1.018659097s"
level=info msg="File cache stats: 64 entries of total size 336.2KiB"
level=info msg="Memory: 690 samples, avg is 237.1MB, max is 815.6MB"
level=info msg="Execution took 1m8.844690913s"
```

With the cache populated:
```bash
└─❯  ./hack/verify-golint.sh
level=info msg="[config_reader] Config search paths: [./ /app / /root]"
level=info msg="[config_reader] Used config file .golangci.yml"
level=info msg="[lintersdb] Active 24 linters: [asciicheck bodyclose deadcode errcheck errorlint exhaustive exportloopref gofmt goimports golint gosec govet ineffassign misspell noctx rowserrcheck scopelint sqlclosecheck staticcheck structcheck typecheck unparam unused varcheck]"
level=info msg="[lintersdb] Active presets: [bugs unused]"
level=info msg="[loader] Go packages loading at mode 575 (types_sizes|compiled_files|deps|imports|exports_file|files|name) took 41.164374105s"
level=info msg="[runner/filename_unadjuster] Pre-built 0 adjustments in 9.253729ms"
level=info msg="[linters context/goanalysis] analyzers took 0s with no stages"
level=info msg="[linters context/goanalysis] analyzers took 0s with no stages"
level=info msg="[runner] Issues before processing: 122, after processing: 0"
level=info msg="[runner] Processors filtering stat (out/in): cgo: 122/122, path_prettifier: 122/122, skip_files: 122/122, autogenerated_exclude: 47/122, filename_unadjuster: 122/122, exclude-rules: 0/47, identifier_marker: 47/47, exclude: 47/47, skip_dirs: 122/122"
level=info msg="[runner] processing took 4.62487ms with stages: identifier_marker: 1.427637ms, path_prettifier: 1.420468ms, autogenerated_exclude: 1.22471ms, skip_dirs: 308.56µs, exclude-rules: 214.768µs, cgo: 15.828µs, filename_unadjuster: 7.258µs, nolint: 1.386µs, max_same_issues: 783ns, skip_files: 573ns, uniq_by_line: 455ns, path_shortener: 341ns, exclude: 341ns, max_from_linter: 339ns, diff: 276ns, source_code: 269ns, severity-rules: 240ns, sort_results: 239ns, path_prefixer: 201ns, max_per_file_from_linter: 198ns"
level=info msg="[runner] linters took 195.810474ms with stages: goanalysis_metalinter: 187.756328ms, unused: 3.292458ms"
level=info msg="File cache stats: 0 entries of total size 0B"
level=info msg="Memory: 415 samples, avg is 71.0MB, max is 137.6MB"
level=info msg="Execution took 41.381169026s"
```

Reply:
I still can't figure out why
> Go packages loading at mode 575 (types_sizes|compiled_files|deps|imports|exports_file|files|name) took 41.164374105s

takes so long.
On subsequent runs, this takes about 20s and I'd like to further get it down.

But, this is a good improvement and we should merge this in.